sqlglot/expressions.py

```diff
@@ -6563,6 +6563,29 @@ def or_(
     return t.cast(Condition, _combine(expressions, Or, dialect, copy=copy, **opts))
 
 
+def xor(
+    *expressions: t.Optional[ExpOrStr], dialect: DialectType = None, copy: bool = True, **opts
+) -> Condition:
```

Comment:
shouldn't this return xor?

Reply:
This follows the conventions of `and_()` and `or_()` which return `Condition` with the combined conjuctions as expression afaict

Reply:
I think this is tech debt, I added the type hints a long time ago. Can you check if we can make them `Or`, `And` and `Xor`?

Reply:
By adjusting both the type hint and the `return t.cast(<type>,  ...)` to the appropriate type I don't see any complaints from style, unit and test. Should I include that refactor here?

Reply:
Yea let's do it

Reply:
Ah, I just remembered why I used `Condition` instead of `Or`.. It's because `or_(1)` works and yields a single literal. So the previous one was good. Apologies for the back and forth @VaggelisD, let's just revert that last change and this should be good to go 👍.